### PR TITLE
Password strength added on signup page

### LINKED ISSUE #544: Add password strength on signup page
**Actual Behaviour**

While entering the password in signup page there is no password strength indicator.

**Expected Behaviour**

While entering the password on signup page the password strength is shown.

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by samagragupta**
Fixes #544 

Changes: While entering the password on signup page the password strength is shown.

Surge Deployment Link: https://pr-555-fossasia-susi-accounts.surge.sh

Screenshots for the change:
signup-strength

### CODE REVIEW

**Comment by samagragupta** [comment]
@rossoskull Done 😄

File: src/components/Auth/SignUp/SignUp.react.js [hunk 1]
```diff
@@ -113,6 +117,15 @@ export default class SignUp extends Component {
       passwordConfirmError = !(
         passwordValue === this.state.confirmPasswordValue
       );
+      if (validPassword) {
+        let result = zxcvbn(passwordValue);
+        state.passwordScore = result.score;
```
Comment (anshumanv) [inline src/components/Auth/SignUp/SignUp.react.js]: State should never be updated this way, see basic react docs.

**Comment by samagragupta** [comment]
@anshumanv @rossoskull Done the required changes.

File: src/components/Auth/SignUp/SignUp.react.js [hunk 2]
```diff
@@ -296,6 +312,7 @@ export default class SignUp extends Component {
     );
 
     const PasswordClass = [`is-strength-${this.state.passwordScore}`];
+    const PasswordClassName = [`is-strength-name-${this.state.passwordScore}`];
```
Comment (fragm3) [inline src/components/Auth/SignUp/SignUp.react.js]: Why are you using array?

File: src/components/Auth/SignUp/SignUp.react.js [hunk 3]
```diff
@@ -364,6 +381,11 @@ export default class SignUp extends Component {
                   textFieldStyle={{ padding: '0px' }}
                 />
                 <div className="ReactPasswordStrength-strength-bar" />
+                <div>
+                  <span className={PasswordClassName.join(' ')}>
```
Comment (fragm3) [inline src/components/Auth/SignUp/SignUp.react.js]: No needed if PasswordClassName is string.

**Comment by samagragupta** [comment]
@fragm3 Done the required changes.

File: src/components/Auth/SignUp/SignUp.css [hunk 4]
```diff
@@ -127,6 +127,19 @@
     color: #fff;
 }
 
+.is-strength-0 { color: #D1462F; }
+.is-strength-1 { color: #D1462F; }
+.is-strength-2 { color: #57B8FF; }
+.is-strength-3 { color: #57B8FF; }
+.is-strength-4 { color: #2FBF71; }
+
+.is-strength-name-0 { margin-left: 44%;}
```
Comment (akshatnitd) [inline src/components/Auth/SignUp/SignUp.css]: Instead of using variable margin for each strength, use margin-auto and align it center to the container

**Comment by samagragupta** [comment]
@akshatnitd @anshumanv @mariobehling Please review.

File: src/components/Auth/SignUp/SignUp.css (hunk 4 above)
Comment (anshumanv) [reply src/components/Auth/SignUp/SignUp.css]: These changes are unattended.

Comment (anshumanv) [reply src/components/Auth/SignUp/SignUp.css]: Make margin as auto

**Pull request merged**